### ISSUE TITLE
'tailor' module continually requests new job for toads

### ISSUE THREAD

**Issue by Stealthii** [DFHack/dfhack#2916]
I'm utilizing the tailor module to keep up with tattered clothing demands in my fort. Recently, I've been seeing the module request new silk trousers sized for toads, despite my stockpile now containing an innumerable number of toad sized trousers.
```
tailor: confiscated rope reed dress from Imush äkilmonom.
tailor: added order #1448 for 1 silk trousers, sized for toads
tailor: civilization cannot make skirts, skipped
tailor: ordered 1 items of clothing
Fixed priority of manager orders.
tailor: confiscated kenaf cloak from Sazir Uzolzulban.
tailor: added order #1449 for 1 silk trousers, sized for toads
tailor: civilization cannot make skirts, skipped
tailor: ordered 1 items of clothing
Fixed priority of manager orders.
tailor: added order #1450 for 1 silk trousers, sized for toads
tailor: civilization cannot make skirts, skipped
tailor: ordered 1 items of clothing
Fixed priority of manager orders.
tailor: added order #1451 for 1 silk trousers, sized for toads
tailor: civilization cannot make skirts, skipped
tailor: ordered 1 items of clothing
Fixed priority of manager orders.
tailor: added order #1452 for 1 silk trousers, sized for toads
tailor: civilization cannot make skirts, skipped
tailor: ordered 1 items of clothing
Fixed priority of manager orders.
```
![image](https://user-images.githubusercontent.com/998920/219454771-3099aaf0-5bf2-4a9b-898a-442c8dc80104.png)

There are no toads or toad men in my fortress, the closest thing I have is a single green tree frog, but no-one owns her as a pet or has tried to put little trousers on her yet, so I'm not sure where this order is being generated from.

**Comment by ab9rf** [comment]
I've added a test to prevent attempting to make clothing for a bodysize that isn't mapped to a race. Toads are race 0 in the default raws, which means an unmapped body size gets mapped to 0 (toads) by default. The code will now check to see if there is a known mapping for a given body size and refuse to order anything if the size is unmapped.

The diagnostic generated when this happens will identify the size that was not mapped correctly, which may help to identify why this is happening in the first place.

Once GHA finishes the build for this PR, if you could test on your save, I'd appreciate it. Could you, using the version in the build artifact from that PR (see https://github.com/DFHack/dfhack/actions/runs/4197987416), do `debugfilter set tailor trace` to set the logging level to trace, then `tailor once` to trigger a processing cycle, and provide the resulting output?

**Comment by Stealthii** [comment]
I've attached [tailor_cycle.txt](https://github.com/DFHack/dfhack/files/10768657/tailor_cycle.txt) with the complete log of tailor run, the following line springs to mind:
```
tailor: cannot determine race for clothing of size 5500, skipped
```

**Comment by ab9rf** [comment]
thank you that's very helpful. are you using any mods? do you have any fort citizens or residents who are not dwarfs, and if so what race are they?

**Comment by ab9rf** [comment]
should be fixed be #2917

**Comment by Stealthii** [comment]
I had a Wolf Woman Bard, Sog Ostararzes, who died in the epic bar fight of '66 following a performance with an artifacted instrument of immense beauty. It was around this time (one year ago) that I started noticing tiny trousers sized for toads appearing in the clothing supply.

Thanks for taking a look at this, and patching out the erroneous order creation.
